builtin/providers/cidr/datasource_cidr_network.go

```diff
@@ -0,0 +1,117 @@
+package cidr
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+type subnetMask struct {
+	Name string
+	Mask int
+}
+
+func dataSourceNetwork() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNetworkRead,
+
+		Schema: map[string]*schema.Schema{
+			"cidr_block": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The CIDR Block for the entire network (aka supernet)",
+				Required:    true,
+			},
+			"subnet": &schema.Schema{
+				Type:        schema.TypeList,
+				Description: "The desired subnet masks to create",
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"mask": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+			"subnet_cidrs": &schema.Schema{
+				Type:        schema.TypeMap,
+				Description: "Map keyed by subnet name returning the value of the network in CIDR notation",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceNetworkRead(d *schema.ResourceData, meta interface{}) error {
+	var masterCIDRList []*net.IPNet
+	rawSubnetMasks := d.Get("subnet").([]interface{})
+	subnetMasks := expandSubnetMasks(rawSubnetMasks)
+	cBlock := d.Get("cidr_block").(string)
+	_, startNet, err := net.ParseCIDR(cBlock)
+	if err != nil {
+		return fmt.Errorf("Error pasring CIDR %v for cidr_network\n", err)
```

Comment:
*parsing*

Reply:
fixed in https://github.com/moofish32/terraform-provider-cidr/pull/2